### ISSUE TITLE
Inspection  is not detected for right curly brace `}` before `else` in a `let...else`

### ISSUE THREAD

**Issue by mili-l** [intellij-rust/intellij-rust#10256]
## Environment

* **IntelliJ Rust plugin version:** 0.4.191.5268-231-nightly
* **Rust toolchain version:** 1.69.0-nightly (bd39bbb4b 2023-02-07) x86_64-apple-darwin
* **IDE name and version:** CLion 2023.1 EAP (CL-231.8109.4)
* **Operating system:** macOS 12.6
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr, org.rust.cargo.emulate.terminal, org.rust.macros.proc

## Problem description
Right curly brace `}` before `else` in a `let...else` statement not allowed

## Steps to reproduce
There is a code
```rust
fn baz() {
    let Some(x) = { Some(1) } else {
        return;
    };
}
```
## Actual Results
 Inspection of right curly brace `}` before `else` is not detected

<img width="698" alt="Screenshot 2023-03-16 at 12 16 17" src="https://user-images.githubusercontent.com/1213122/225600752-aa37be57-bf8f-4049-a152-b1e1f1aefa90.png">

Related to #7798
